service/matching/matching_engine.go

```diff
@@ -615,6 +615,10 @@ pollLoop:
 		resp, err := e.recordWorkflowTaskStarted(ctx, requestClone, task)
 		if err != nil {
 			switch err.(type) {
+			case *serviceerror.Internal, *serviceerror.DataLoss:
+				e.nonRetryableErrorsDropTask(task, taskQueueName, err)
```

Comment:
I would just inline the whole thing, like the other clauses. but fine either way

Reply:
feel like it might get too clustered + the other cases also give that notion right now. Since we are also emitting a metric in this case, let's go ahead with keeping the func call?

Reply:
some refactoring with common tags might help, but no worries